Duplicati/Library/Utility/WinTools.cs

```diff
@@ -53,5 +55,110 @@ public static string GetWindowsGpgExePath()
                 "win-tools", "gpg.exe");
             return string.IsNullOrEmpty(wintoolsPath) ? null : wintoolsPath;
         }
+
+
+        /// <summary>
+        /// On Windows platforms, tries to add <see cref="Win32.SE_BACKUP_NAME"/> privilege to the process
+        /// before executing the specified <paramref name="action"/>.
+        /// Executes the specified <paramref name="action"/> as-is if the additional process privilege cannot be added
+        /// to the process security context (eg current user is not allowed to enable this privilege).
+        /// </summary>
+        /// <param name="action">Action to execute.</param>
+        /// <remarks>
+        /// In Windows platforms, the <see cref="Win32.SE_BACKUP_NAME"/>
+        /// privilege allows users to bypass file and directory object permissions for the purposes of backing up the system.
+        /// 
+        /// The privilege is available to user accounts (and security groups) specified in the 
+        /// 'Back up files and directories' group policy (GPO =&gt; Computer Configuration =&gt;
+        /// Windows Settings =&gt; Security Settings =&gt; Local Policies =&gt; User Rights Assignment).
+        /// 
+        /// By default, the 'Administrators' and 'Backup Operators' security groups are specified
+        /// in the 'Back up files and directories' group policy.
+        /// 
+        /// To take advantage of this functionality, configure the Duplicati service to use its own dedicated user account,
+        /// and add this user account to the 'Backup Operators' security group. The process will be able to access ALL files on the system,
+        /// without any other admin permissions.
+        /// </remarks>
+        public static void WithBackupPrivileges(Action action)
+        {
+            if (Platform.IsClientWindows)
+            {
+                var addedPrivilege = AddPrivilege(Win32.SE_BACKUP_NAME);

```

Comment:
Can `AddPrivilege` throw an exception?  If so, I'm not sure if we want to halt the operation, or log a warning and continue without privileges.  Currently, if we encounter a file that we cannot read, I think we skip it and add a log entry.

The latter option might have implications where the retention policy may delete a "privileged" backup in favor for a "non-privileged" one.

Reply:
`AddPrivilege` should never throw.
Logging when privilege enablement occurs (success or failure) is a good idea. I think `warning` severity is too high - it is not a requirement to have this privilege. Ultimately, whether the privilege is available or not does not dictate if reading files will succeed/fail. The process may have full R/W permissions to the files even without this privilege.